raimitigations/dataprocessing/imputer/knn_imputer.py

```diff
@@ -152,11 +152,13 @@ def _fit(self):
         self.imputer.fit(df_valid)
 
     # -----------------------------------
-    def _transform(self, df: pd.DataFrame):
+    def _transform(self, df: pd.DataFrame) -> Union[pd.DataFrame, np.ndarray]:
```

Comment:
The ``_transform()`` method for both the knn_imputer and iterative_imputer are almost identical (the only change is that in one you access the internal variable ``self.knn_params``, and in the other you access ``self.iterative_params``. Maybe you can modularize this part as well by creating a method in the parent class that does the bulk of the job.

Reply:
yes! now it's pre_process(), transform(), post_process(). Thank you!